lib/fortune.js

```diff
@@ -316,8 +339,43 @@ Fortune.prototype.resource = function(name, schema, options, schemaCallback) {
     authMethods: authMethods,
     validation: validation
   };
+
   plugins.init(this, this._resources[name]);
+
   hooks.initGlobalHooks(_this._resources[name], _this.options);
+
+  var customHooks = _.flatten(_.map(_.keys(customTypes), function(key) {
+    // Modify a names of the hooks to include field name they applied to
+    _.each(customTypes[key].hooks, function(whenHooks, when) {
```

Comment:
Usage example, just in case: https://bitbucket.org/flyvictor/user-service/pull-requests/853/sup-661-custom-type-usage-example-do-not/diff